typespec-tests/src/samples/java/com/cadl/armresourceprovider/generated/TopLevelArmResourceInterfaceActionSamples.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) TypeSpec Code Generator.
+
+package com.cadl.armresourceprovider.generated;
+
+/**
+ * Samples for TopLevelArmResourceInterface Action.
+ */
+public final class TopLevelArmResourceInterfaceActionSamples {
+    /*
+     * x-ms-original-file: 2023-11-01/TopLevelArmResourceInterface_Action.json
```

Comment:
Here is what path would look like, for typespec

The combination with tsp-location.yaml directory would be in automation job
https://github.com/Azure/azure-sdk-tools/pull/8854